src/main/java/ro/unibuc/triplea/infrastructure/game/repository/GameRepositoryImpl.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Override
+    public List<GameResponse> findAll(Optional<Integer> count) {
+        String apiUrl = "http://api.steampowered.com/ISteamApps/GetAppList/v0002/?key=" + steamApiKey + "&format=json";
+        ResponseEntity<JsonNode> responseEntity = restTemplate.getForEntity(apiUrl, JsonNode.class);
+
+        if (responseEntity.getStatusCode().is2xxSuccessful()) {
+
+            JsonNode responseNode = responseEntity.getBody();
+
+            List<GameResponse> games = new ArrayList<>();
+            if (responseNode != null && responseNode.has("applist")) {
+                JsonNode appList = responseNode.get("applist");
+                if (appList.has("apps")) {
+                    JsonNode apps = appList.get("apps");
+
+                    int limit = count.orElse(Integer.MAX_VALUE);
+                    int counter = 0;
+                    for (JsonNode app : apps) {
+                        if (counter >= limit) {
+                            break;
+                        }
+
+                        if (app.get("name").asText() != "") {
+                            GameResponse game = GameResponse.builder().gameSteamId(app.get("appid").asInt()).gameName(app.get("name").asText()).build();
+                            games.add(game);
+
+                            counter++;
+                        }
+                    }
+                }
+            }
+
+            return games;
+        } else {
+            System.err.println("Failed to fetch games from the external API. Status code: " + responseEntity.getStatusCode());
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public Game save(Game game) {
+        return springDataGameRepository.save(game);
+    }
+
+    @Override
+    public Optional<GameResponse> findByGameSteamId(Integer gameSteamId) {
+        String apiUrl = "http://api.steampowered.com/ISteamApps/GetAppList/v0002/?key=" + steamApiKey + "&format=json";
+        ResponseEntity<JsonNode> responseEntity = restTemplate.getForEntity(apiUrl, JsonNode.class);
+
+        if (responseEntity.getStatusCode().is2xxSuccessful()) {
+            JsonNode responseNode = responseEntity.getBody();
+
+            if (responseNode != null && responseNode.has("applist")) {
+                JsonNode appList = responseNode.get("applist");
+                if (appList.has("apps")) {
+                    JsonNode apps = appList.get("apps");
+                    for (JsonNode app : apps) {
+                        if (app.get("appid").asInt() == gameSteamId) {
+                            GameResponse game = GameResponse.builder().gameSteamId(app.get("appid").asInt()).gameName(app.get("name").asText()).build();
+                            return Optional.of(game);
+                        }
+                    }
+                }
+            }
+        } else {
+            System.err.println("Failed to fetch games from the external API. Status code: " + responseEntity.getStatusCode());
```

Comment:
Do not use System.err as logging. Use @Slf4j or another logging alternative (log4j , etc.)

Reply:
used slf4j